lib/mllp/receiver.ex

```diff
@@ -214,6 +196,44 @@ defmodule MLLP.Receiver do
     %{num_acceptors: 100, max_connections: 20_000}
   end
 
+  defp update_transport_options(transport_opts, port) do
+    {transport_module, tls_options1, transport_opts1} =
+      case Map.pop(transport_opts, :tls) do
+        {nil, options1} ->
+          Logger.warn(
+            "Starting listener on a non secured socket, data will be passed over unencrypted connection!"
+          )
+
+          {:ranch_tcp, [], options1}
+
+        {tls_options, options1} ->
+          {:ranch_ssl, Keyword.merge(default_tls_options(), tls_options), options1}
+      end
+
+    socket_opts =
+      transport_opts
+      |> get_socket_options(port)
```

Comment:
This could be  reduced to simply `socket_opts = get_socket_options(transport_opts, port) ++ tls_options`  and then`mergee_with_tls_options/2`could be removed.